pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +281,56 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) netContainerCleanup(containerInfo containerGCInfo) {
```

Comment:
> and all dead containers are left to GC

It wouldn't have to cleanup _all_ dead containers.. just any dead container that is relevant to the pod being synced (i.e a previous "pause" container).  Do we think that cleaning up a single container will introduce too much latency?  Also I would think this shouldn't introduce any latency in mainline "I launched a new pod" type scenarios, this should only happen when a running pod is killed by something external (e.g docker / the node restarting)